packages/astro/src/assets/internal.ts

```diff
@@ -100,13 +158,23 @@ export async function getImage(
 		: [];
 
 	let imageURL = await service.getURL(validatedOptions, imageConfig);
+
+	const matchesOriginal = (transform: ImageTransform) =>
```

Comment:
In theory, couldn't the `quality` be different here? Or any other parameters that affects the image, for image services who support more.

Reply:
I think in this scenario it can't, as this is just checking if it's the srcset entry which is the same as the src. I don't think we expose a way to set this separately.

Reply:
Can't an image service set a different quality for a srcset transform?

Reply:
I guess in theory. I'm comfortable with not supporting that though. It's a terrible idea.